### PR TITLE
[contrib] make func_name work on any callable, not only funcs

### PR DESCRIPTION
**Pull request by ufoot**

### CODE REVIEW

File: tests/contrib/test.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
```
Comment (palazzem) [inline tests/contrib/test.py]: I think we can easily reproduce the issue adding a Django view that doesn't work if the name is collected as before.

File: tests/contrib/test.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    return "nothing"
+
+
+class TestContrib(object):
+    """
+    Testing contrib helper funcs.
+    """
+
+    def test_func_name(self):
```
Comment (clutchski) [inline tests/contrib/test.py]: can you test this with static and class methods?

**Comment by palazzem** [comment]
@ufoot any updates for this one?

File: tests/contrib/test.py (hunk 1 above)
Comment (palazzem) [inline tests/contrib/test.py]: solved here: https://github.com/DataDog/dd-trace-py/pull/195/files#diff-7ed2e71e60947f800107456ca4af05aeR52

File: tests/contrib/test.py (hunk 2 above)
Comment (palazzem) [inline tests/contrib/test.py]: Tested here: https://github.com/DataDog/dd-trace-py/pull/195/files#diff-e95dd2789103d2810b8758d53801377fR38

Django use case has been tested here: https://github.com/DataDog/dd-trace-py/pull/195/files#diff-7ed2e71e60947f800107456ca4af05aeR52

File: tests/contrib/django/test_middleware.py [hunk 3]
```diff
@@ -47,6 +47,35 @@ def test_middleware_trace_errors(self):
         span = spans[0]
         eq_(span.get_tag('http.status_code'), '403')
         eq_(span.get_tag('http.url'), '/fail-view/')
+        eq_(span.resource, 'tests.contrib.django.app.views.ForbiddenView')
```
Comment (ufoot) [inline tests/contrib/django/test_middleware.py]: 👍

File: tests/contrib/django/app/views.py [hunk 4]
```diff
@@ -23,10 +26,34 @@ def get(self, request, *args, **kwargs):
         return HttpResponse(status=403)
 
 
+def function_view(request):
+    return HttpResponse(status=200)
+
+
+class FeedView(Feed):
```
Comment (ufoot) [inline tests/contrib/django/app/views.py]: OK, that was the missing bit I couldn't figure out not knowing Django enough I guess. Thanks!

Comment (palazzem) [inline tests/contrib/django/app/views.py]: Yes! removing the fix, will reproduce the error we had before. I could use a plain CBV with ``__call__``, but that one is a real use case

**Pull request merged**